Kernel/Devices/HID/PS2MouseDevice.cpp

```diff
@@ -139,70 +139,69 @@ MousePacket PS2MouseDevice::parse_data_packet(const RawPacket& raw_packet)
     return packet;
 }
 
-u8 PS2MouseDevice::get_device_id()
+ErrorOr<u8> PS2MouseDevice::get_device_id()
 {
-    if (send_command(I8042Command::GetDeviceID) != I8042Response::Acknowledge)
-        return 0;
+    TRY(send_command(I8042Command::GetDeviceID));
     return read_from_device();
 }
 
-u8 PS2MouseDevice::read_from_device()
+ErrorOr<u8> PS2MouseDevice::read_from_device()
 {
     return m_i8042_controller->read_from_device(instrument_type());
 }
 
-u8 PS2MouseDevice::send_command(u8 command)
+ErrorOr<u8> PS2MouseDevice::send_command(u8 command)
 {
-    u8 response = m_i8042_controller->send_command(instrument_type(), command);
-    if (response != I8042Response::Acknowledge)
+    u8 response = TRY(m_i8042_controller->send_command(instrument_type(), command));
+
+    if (response != I8042Response::Acknowledge) {
         dbgln("PS2MouseDevice: Command {} got {} but expected ack: {}", command, response, static_cast<u8>(I8042Response::Acknowledge));
+        // FIXME: Is this the correct errno value for this?
```

Comment:
It's probably the right errno, but let other developers to look at this and to check this.